### ISSUE TITLE
[IOS] After call getAvailablePurchases(), purchaseUpdatedListener triggered many times

### ISSUE THREAD

**Issue by ng-ha** [hyochan/react-native-iap#2815]
in IOS, every time i call getAvailablePurchases(), purchaseUpdatedListener triggered many times, when there is no new purchases. How do i get available purchases or purchases from history without triggering purchaseUpdatedListener?

I tried following the example, but i found this:       [isConsumable: true,](https://github.com/dooboolab-community/react-native-iap/blob/1280cd9b970c77af170835c1e915da2234526013/IapExample/src/screens/Subscriptions.tsx#L68) when finish a SUBSCRIPTION transaction. Is this intended?

**Comment by cagdaseksi** [comment]
@ng-ha I'm experiencing the same issue. Could you share the code inside the purchaseUpdatedListener?

<img width="724" alt="Screenshot 2024-08-22 at 17 28 10" src="https://github.com/user-attachments/assets/2e0df09f-b693-4a18-a163-111b54e1f49b">

**Comment by ng-ha** [comment]
@cagdaseksi i fixed it by using Storekit2: `setup({storekitMode: 'STOREKIT2_MODE'});`

**Comment by NevilPatel01** [comment]
I would suggest not to use UseEffect for every function instead use single useEffect and add all required function in it just to make sure everything goes step by step and don't break any thing in between. And as per documentation they suggest adding `purchaseUpdatedListener` when application start. 'It is recommended that you start listening to updates as soon as your application launches' [Check full detial here](https://react-native-iap.dooboolab.com/docs/guides/purchases).

**Comment by tieumainhi** [comment]
Can u handle it?

**Comment by nthanhha** [comment]
@Tieu-Mai-Nhi yes. I fixed it by using Storekit2: setup({storekitMode: 'STOREKIT2_MODE'});
